hypothesis-python/src/hypothesis/core.py

```diff
@@ -1116,11 +1116,10 @@ def find(
     # type: (...) -> Any
     """Returns the minimal example from the given strategy ``specifier`` that
     matches the predicate function ``condition``."""
-    settings = settings or Settings(
-        max_examples=2000,
-        max_shrinks=2000,
+    settings = Settings(
+        settings or Settings(max_examples=2000),
+        suppress_health_check=HealthCheck.all()
```

Comment:
OK, the `settings` argument is either (a) `None`, or (b) a `settings` instance (here aliased to `Settings`).  We are aiming to reassign `settings` to a new `Settings` instance with all health checks suppressed but otherwise using the passed in settings *unless* that was `None` in which case we want `max_examples=2000` with the *other* settings at their global default.

This will probably be clearer as two one-line assignments, which are coming up.